bin/controller.py

```diff
@@ -38,22 +39,34 @@ def post_new():
         raise bt.HTTPError(411, "Content-Length required")
     if int(content_length) > config.SNIPPET_MAX_SIZE:
         raise bt.HTTPError(413, f"Payload too large, we accept maximum {config.SNIPPET_MAX_SIZE} bytes")
+    
+    files = bt.request.files
+    forms = bt.request.forms
 
     code = None
-    if bt.request.files:
-        code = next(bt.request.files.values()).file.read(config.SNIPPET_MAX_SIZE)
-    elif bt.request.forms:
-        code = bt.request.forms.get('code', '').encode('latin-1')
+    if files:
+        code = next(files.values()).file.read(config.SNIPPET_MAX_SIZE)
+    elif forms:
+        code = forms.get('code', '').encode('latin-1')
     if not code:
         raise bt.HTTPError(417, "Missing code")
 
     ext = ""
-    if bt.request.forms:
-        lang = bt.request.forms.get('lang')
+    if forms:
+        lang = forms.get('lang')
         if lang:
             ext = "."+langtoext.get(lang, lang)
 
-    snippet = Snippet.create(code.decode('utf-8'))
+    max_views = math.inf
+    if not forms.get('infinite_views'):
+        try:
+            max_views = int(forms.get('max_views'))
```

Comment:
Il serait bien d'utiliser une valeur par défaut ici (comme +infini)
```suggestion
            max_views = int(forms.get('max_views', '+inf'))
```

Reply:
Il faut utiliser `float`, `+inf` n'est pas un entier valide

Reply:
La valeur par défaut est déjà en haut : `max_views = math.inf`. Je changerai pour `float('+inf')` comme à dit lazor plus tard

Reply:
Malheureusement `forms.get("max_view")` peut renvoyer None; et donc max_view sera None

Reply:
Ah j'ai oublié de changer ça...